packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -366,15 +361,45 @@ export function TransferPanelMain({
   const { walletAddress } = arbTokenBridge
 
   const {
-    eth: [ethL1Balance]
-  } = useBalance({ provider: l1.provider, walletAddress })
+    eth: [ethL1Balance],
+    erc20: [erc20L1Balances]
+  } = useBalance({
+    provider: l1.provider,
+    walletAddress
+  })
   const {
-    eth: [ethL2Balance]
-  } = useBalance({ provider: l2.provider, walletAddress })
+    eth: [ethL2Balance],
+    erc20: [erc20L2Balances]
+  } = useBalance({
+    provider: l2.provider,
+    walletAddress
+  })
 
   const isSwitchingL2Chain = useIsSwitchingL2Chain()
 
-  const tokenBalances = useTokenBalances(selectedToken?.address)
+  const selectedTokenBalances = useMemo(() => {
+    const result: {
+      [NetworkType.l1]: BigNumber | null
+      [NetworkType.l2]: BigNumber | null
+    } = {
+      l1: null,
+      l2: null
+    }
+
+    if (!selectedToken) {
+      return result
+    }
+
+    if (erc20L1Balances) {
+      result.l1 = erc20L1Balances[selectedToken.address] || null
+    }
+
+    if (erc20L2Balances && selectedToken.l2Address) {
+      result.l2 = erc20L2Balances[selectedToken.l2Address] || null
+    }
```

Comment:
Hm, in the previous implementation we defaulted to `constants.zero`? Also nit: `??` instead of `||`

Reply:
Yeah I think it's better to leave it as it was